core/touch_gesture.js

```diff
@@ -0,0 +1,287 @@
+/**
+ * @license
+ * PXT Blockly
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * https://github.com/Microsoft/pxt-blockly
+ *
+ * See LICENSE file for details.
+ */
+
+/**
+ * @fileoverview The class extends Blockly.Gesture to support pinch to zoom
+ * for both pointer and touch events.
+ * @author samelh@microsoft.com (Sam El-Husseini)
+ */
+'use strict';
+
+goog.provide('Blockly.TouchGesture');
+
+goog.require('Blockly.Gesture');
+
+goog.require('goog.asserts');
+goog.require('goog.math.Coordinate');
+
+
+/*
+ * Note: In this file "start" refers to touchstart, mousedown, and pointerstart
+ * events.  "End" refers to touchend, mouseup, and pointerend events.
+ */
+
+/**
+ * Class for one gesture.
+ * @param {!Event} e The event that kicked off this gesture.
+ * @param {!Blockly.WorkspaceSvg} creatorWorkspace The workspace that created
+ *     this gesture and has a reference to it.
+ * @constructor
```

Comment:
Missing annotation: `@extends {Blockly.Gesture}`

Within the annotation the whole class can disappear during compression.

Reply:
Good to know.